go/test/endtoend/onlineddl/vrepl_suite/onlineddl_vrepl_suite_test.go

```diff
@@ -161,7 +182,14 @@ func readTestFile(t *testing.T, testName string, fileName string) (content strin
 
 // testSingle is the main testing function for a single test in the suite.
 // It prepares the grounds, creates the test data, runs a migration, expects results/error, cleans up.
-func testSingle(t *testing.T, testName string) {
+func testSingle(t *testing.T, testName string, fkOnlineDDLPossible bool) {
+	if _, exists := readTestFile(t, testName, "require_rename_table_preserve_foreign_key"); exists {
+		if !fkOnlineDDLPossible {
+			t.Skipf("Skipping test due to require_rename_table_preserve_foreign_key")
+			return
+		}
+	}
```

Comment:
Can we instead use `mysqlctl.GetVersionString()` here and check for the PS fork indicator in the string?

Reply:
That's what https://github.com/vitessio/vitess/pull/15376/files#diff-5b2f7e26e6a031b4098461528dca98e5ff630e70e1986f35efaab59f3fa5bd76R137-R156 does (not using `version()` but using variables). The intent of the flag file is to point out that this specific test should only run where  `preserve_foreign_key` is supported. We can't automatically associate a specific test with this specific restriction.

Reply:
Run it